src/chart/boxplot/BoxplotView.ts

```diff
@@ -180,17 +183,76 @@ function updateNormalBoxData(
 
     el.useStyle(data.getItemVisual(dataIndex, 'style'));
     el.style.strokeNoScale = true;
-
     el.z2 = 100;
 
     const itemModel = data.getItemModel<BoxplotDataItemOption>(dataIndex);
     const emphasisModel = itemModel.getModel('emphasis');
 
+    if (seriesModel.option.label.show) {
+        const formattedLabels =
+            ((seriesModel as BoxplotSeriesModel).getRawValue(dataIndex) as number[])
+            .splice(1).map((value: number) => seriesModel.option.label.formatter(value).toString());
```

Comment:
You should probably not create a `Text` element and do all the stuffs by yourself. Instead, `setLabelStyle` should be called. See how this is done with [line series](https://github.com/apache/echarts/blob/master/src/chart/helper/Line.ts#L275-L289). But in boxplot's case, there are multiple labels so you need to figure out how this should be done, but generally speaking, `setLabelStyle` should be called most likely.

Reply:
I would not add it to `setLabelStyle` as it is super specific code which only applies to the bar chart, so it doesn't really make sense in my opinion to add it a general util method. However, I agree that it should live in the `labelStyle` util. So I would vote to move it there and refactor it so it does not create a Text element manually. What do you think?

Reply:
@Ovilia any news on this? Is the proposed solution good enough?